### PR TITLE
Add more to dds

### PR DESCRIPTION
**Pull request by vicancy**
Makes the schema definition more general, not constraint to DocFX implementation.

### CODE REVIEW

File: Documentation/spec/docfx_document_schema.md [hunk 1]
```diff
@@ -1,126 +1,160 @@
...
+### 6.4 d-tags
+The value of this keyword MUST be an `array`, elements of the array MUST be strings and MUST be unique. It provides hints for applications to decide how to interpret the property, for example, `localizable` tag can help Localization team to interpret the property as *localizable*; `metadata` tag can help DocFX to fill in additional metadata, e.g. github commit information.
 
-The extension keywords are RECOMMENDED to has the form `{an charactor other than "d"}-{keyword}`, so that it can be easily distinguished from keywords defined in this schema and JSON schema. They can have any valid JSON format value.
+### 6.5 d-merge
+The value of this keyword MUST be a string. It specifies how to merge two values of the given property. One use scenario is how DocFX uses the [overwrite files](..\tutorial\intro_overwrite_files.md) to overwrite the existing values. In the below table, we use `source` and `target` to stands for the two values for merging.
 
-## 8. Samples
+The value MUST be one of the following:
+| Value      | Description
```
Comment (superyyrrzz) [inline Documentation/spec/docfx_document_schema.md]: add a blank line above

File: Documentation/spec/docfx_document_schema.md [hunk 2]
```diff
@@ -1,126 +1,160 @@
...
-* `includeMarkdownFile`: This is same as `includeFile`, in spite of that the content will be marked up before filled in.
+| Field Name      | Type   | Description
+|-----------------|--------|----------
+| title        | string | The title of the property.
+| description  | string | A lengthy explanation about the purpose of the data described by the schema
+| default      | what `type` defined | The default value for current field
+| type         | string | The type of the root document model. Refer to [type keyword](#6-1-type) for detailed description.
+| properties   | [Property Definitions Object](#property-definitions-object) | An object to hold the schema of all the properties if `type` for the model is `object`. Omitting this keyword has the same behavior as an empty object.
+| items        | [Property Object](#property-object) | An object to hold the schema of the items if `type` for the model is `array`. Omitting this keyword has the same behavior as an empty schema.
+| d-include    | string | Defines the included file path. Refer to [d-contentType](6-2-d-include) for detailed explanation.
```
Comment (vwxyzh) [inline Documentation/spec/docfx_document_schema.md]: `#6-2-d-include`

File: Documentation/spec/docfx_document_schema.md [hunk 3]
```diff
@@ -204,4 +247,13 @@ Here's the schema to describe these operations:
         }
     }
 }
-```
\ No newline at end of file
+```
+
+## 8. Open Issues
+1. DocFX fills `_global` metadata into the processed data model, should the schema reflect this behavior?
```
Comment (superyyrrzz) [inline Documentation/spec/docfx_document_schema.md]: prefer not to put in this schema for brevity. Another doc can describe this.

File: Documentation/spec/docfx_document_schema.md [hunk 4]
```diff
@@ -1,126 +1,160 @@
...
-Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
+| Field Name      | Type   | Description
+|-----------------|--------|----------
+| d-schemaVersion | string | `*`The version of the schema specification
+| d-fileType      | string | `*`The file type of the document model. Supported values are `YAML` and `JSON`.
+| type            | string | `*`The type of the root document model MUST be `object`
+| d-mimeType      | string | The mime type of the document model that applies to current schema. If not set, the schema name will be considered as its mime type.
+| d-info          |[Info Object](#info-object) | `*`The information about current schema
+| properties      | [Property Definitions Object](#property-definitions-object) | An object to hold the schema of all the properties if `type` for the model is `object`
+| items           | [Item Definitions Object](#item-definitions-object) | An object to hold the schema of all the items if `type` for the model is `array`
```
Comment (superyyrrzz) [inline Documentation/spec/docfx_document_schema.md]: >#item-definitions-object [](start = 46, length = 24)

bookmark doesn't exist

File: Documentation/spec/docfx_document_schema.md [hunk 5]
```diff
@@ -1,126 +1,160 @@
-# DocFX Document Schema Design Specification
+# DocFX Document Schema v1.0 Specification
 
 ## 1. Introduction
 DocFX supports different [document processors](..\tutorial\howto_build_your_own_type_of_documentation_with_custom_plug-in.md) to handle different kinds of input. For now, if the data model changes a bit, a new document processor is needed, even most works in processors are the same.
```
Comment (hellosnow) [inline Documentation/spec/docfx_document_schema.md]: >most works [](start = 245, length = 10)

most of the work?

File: Documentation/spec/docfx_document_schema.md [hunk 6]
```diff
@@ -1,126 +1,160 @@
...
-## 7. Extension Keywords
+| Value      | Description
+|------------|-------------
+| `uid`      | `type` MUST be `string`. It means the property defines a unique identifier inside current document model
+| `fileLink` | `type` MUST be `string`. It means the property defines a file link inside current document model. Application CAN help to validate if the linked file exists, and update the file link if the linked file changes its output path.
+| `uidLink`  | `type` MUST be `string`. It means the property defines a UID link inside current document model. Application CAN help to validate if the linked UID exists, and resolve the UID link to the corresponding file output path.
+| `markdown` | `type` MUST be `string`. It means the property is in [DocFX flavored Markdown](..\spec\docfx_flavored_markdown.md) syntax. Application CAN help to transform it into HTML format.
 
-While this schema defines some keywords for DocFX use, additional keywords can be added above it. 
+### 6.4 d-tags
```
Comment (hellosnow) [inline Documentation/spec/docfx_document_schema.md]: >tags [](start = 10, length = 4)

need we list what tags we're going to support? following content type and merge?

Comment (vicancy) [inline Documentation/spec/docfx_document_schema.md]: tags name can be customized and not constrained to a specific range...

File: Documentation/spec/docfx_document_schema.md [hunk 7]
```diff
@@ -204,4 +248,27 @@ Here's the schema to describe these operations:
...
+
+## 8. Open Issues
+1. DocFX fills `_global` metadata into the processed data model, should the schema reflect this behavior?
+   * If YES: 
+        * Pros:
+            1. Users are aware of the existence of `_global` metadata, they can overwrite the property if they want.
+            2. Template writers are aware of it, they can completely rely on the schema to write the template.
+        * Cons:
+            1. Schema writers need aware of the existence of `_global` metadata, it should always exists for any schema. (Should we introduce in a concept of base schema?)
+<<<<<<< HEAD
```
Comment (hellosnow) [inline Documentation/spec/docfx_document_schema.md]: Resolve the conflict...

File: Documentation/spec/docfx_document_schema.md (hunk 3 above)
Comment (herohua) [inline Documentation/spec/docfx_document_schema.md]: `_global` metadata is one of the outputs of processor, correct? If so, a separate doc looks better to me. Or, shall we consider also have a schema for outputs?

File: Documentation/spec/docfx_document_schema.md [hunk 8]
```diff
@@ -1,126 +1,161 @@
...
+It defines how applications interpret the property. The values MUST be one of the following:
 
-> [!Note]
-> After applying the overwrite files, the model should be checked again to ensure that all the validation rules still satisfy.
+| Value      | Description
+|------------|-------------
+| `uid`      | `type` MUST be `string`. It means the property defines a unique identifier inside current document model
+| `fileLink` | `type` MUST be `string`. It means the property defines a file link inside current document model. Application CAN help to validate if the linked file exists, and update the file link if the linked file changes its output path.
+| `uidLink`  | `type` MUST be `string`. It means the property defines a UID link inside current document model. Application CAN help to validate if the linked UID exists, and resolve the UID link to the corresponding file output path.
+| `markdown` | `type` MUST be `string`. It means the property is in [DocFX flavored Markdown](..\spec\docfx_flavored_markdown.md) syntax. Application CAN help to transform it into HTML format.
```
Comment (herohua) [inline Documentation/spec/docfx_document_schema.md]: How can we describe `includeFile` and `includeMarkdownFile` in this new proposal?

Comment (vicancy) [inline Documentation/spec/docfx_document_schema.md]: A separate field `d-include`

File: Documentation/spec/docfx_document_schema.md [hunk 9]
```diff
@@ -204,4 +248,24 @@ Here's the schema to describe these operations:
...
+        * Pros:
+            1. Users are aware of the existence of `_global` metadata, they can overwrite the property if they want.
+            2. Template writers are aware of it, they can completely rely on the schema to write the template.
+        * Cons:
+            1. Schema writers need aware of the existence of `_global` metadata, it should always exists for any schema. (Should we introduce in a concept of base schema?)
+    * Decision: *NOT* include, this schema is for **general purpose**, use documents to describe the changes introduced by DocFX.
+2. Is it necessary to prefix `d-` to every field that DocFX introduces in?
+    * If keep `d-`
+        * Pros:
+            1. `d-` makes it straightforward that these keywords are introduced in by DocFX
```
Comment (superyyrrzz) [inline Documentation/spec/docfx_document_schema.md]: >in [](start = 80, length = 2)

remove

**Comment by superyyrrzz** [comment]
:shipit:

File: Documentation/spec/docfx_document_schema.md (hunk 9 above)
Comment (superyyrrzz) [inline Documentation/spec/docfx_document_schema.md]: >in [](start = 80, length = 2)

remove?

**Pull request merged**